src/pykiso/config_parser.py

```diff
@@ -162,20 +162,35 @@ def _fix_types_loc(thing: dict) -> dict:
         return thing
 
     def _parse_env_var(config_element: str) -> str:
-        """Search for an environment variable and replace it with the associated value.
+        """Search for an environment variable and replace it with the associated value if it find one.
+        Otherwise use the default value if one is specified.
 
         Additionally, cast the value if it matches an integer.
 
         :param config_element: config sub-dict value
 
         :return: config sub-dict value with replaced environment variable if needed
         """
+        has_default_value = False
+        # Handle default value
+        if all(element in config_element for element in ["ENV", "="]):
+            default_value = re.search("=(.*)}", config_element).group(1)
+            config_element = config_element[: config_element.find("=")] + "}"
+            has_default_value = True
         # Check for regular expression "ENV{word}"
         match = re.compile(r"ENV{(\w+)}").findall(str(config_element))
```

Comment:
```suggestion
        match_single_env = re.compile(r"ENV{(\w+)}").findall(str(config_element))
        match_env_with_val = re.compile(r"ENV{(\w+)=(.+)}").findall(str(var2)) # search for the pattern
        
        
        
```

Reply:
I even thing it may be feasible to group them in one commando with '|' but I'm not good enough to find the right regex

Reply:
you mean  
(\w+)|=(.+) 
?? ;D
The check if matching group 2 exist...

Reply:
or maybe just this
ENV{(\w+)(=(.+))?}

Reply:
Have you tried them? (the second is better) I struggled on my side to match correctly the 2 possibilities

Reply:
Modified using ENV{(\w+)(=(.+))?}